### PR TITLE
Adpot QDQFinalCleanupTransformer for Q->DQs/DQ->Qs cases

### PR DESCRIPTION
**Pull request by yihonglyu**

### CODE REVIEW

File: onnxruntime/core/optimizer/qdq_transformer/qdq_final_cleanup.cc [hunk 1]
```diff
@@ -31,98 +31,105 @@
...
+      if (input_edge) {
+        graph.AddEdge(src_node_idx, downstream_node_idx, src_arg_idx, downstream_arg_idx);
+      }
     } else {
-      // add Identity node to connect the graph input or initializer to the graph output.
-      Node& id_node = graph.AddNode(graph.GenerateNodeName("QDQFinalCleanupTransformer"),
-                                    "Identity", "", {first_node.MutableInputDefs()[0]}, {graph_output_nodearg});
-      id_node.SetExecutionProviderType(second_node.GetExecutionProviderType());
+      NodeArg* graph_output_nodearg = second_node.MutableOutputDefs()[0];
+      if (src_arg_idx >= 0 && second_node_ptrs.size() == 1) {
```
Comment (yufenglee) [inline onnxruntime/core/optimizer/qdq_transformer/qdq_final_cleanup.cc:116]: there a corner case: Q->{DQ1->output1, DQ2->output2}. there will be 2 different graph output names. It is quite rare though.

File: onnxruntime/test/optimizer/qdq_transformer_test.cc [hunk 2]
```diff
@@ -3725,19 +3727,127 @@
...
-  test_case({{1, 2, 4}, {1, 3, 4}}, false, true, true);   // Block removal of last dq
+  test_case({{1, 2, 4}, {1, 3, 4}}, true, false, true);   // Block removal of last dq
+  test_case({{1, 2, 4}, {1, 3, 4}}, false, true, true);   // Block removal of first dq
   test_case({{1, 2, 4}, {1, 3, 4}}, true, true, true);    // Block removal of first and last dq
 #endif
 }
 
+// test removal of Q->DQs pairs by QDQFinalCleanupTransformer
+TEST(QDQTransformerTests, QDQFinalCleanupTransformer_BasicQDQsCleanup) {
+  auto test_case = [&](bool is_input_q,
```
Comment (yufenglee) [inline onnxruntime/test/optimizer/qdq_transformer_test.cc:3745]: could you please also add a DQ->Qs test case?

File: onnxruntime/core/optimizer/qdq_transformer/qdq_final_cleanup.cc (hunk 1 above)
Comment (yihonglyu) [reply onnxruntime/core/optimizer/qdq_transformer/qdq_final_cleanup.cc:116]: The `QDQFinalCleanupTransformer_BasicQDQsCleanup` covers this when the parameter  `is_dq_output = True`.

File: onnxruntime/test/optimizer/qdq_transformer_test.cc (hunk 2 above)
Comment (yihonglyu) [reply onnxruntime/test/optimizer/qdq_transformer_test.cc:3745]: Added

File: onnxruntime/test/optimizer/qdq_transformer_test.cc [hunk 3]
```diff
@@ -3675,7 +3675,9 @@ TEST(QDQTransformerTests, QDQFinalCleanupTransformer_BasicQDQCleanup) {
     };
 
     // if we block removal of the DQ node the Q node in the pair will not be removed either
-    const int expected_qdq_count = 0 + (block_removal_of_first_dq ? 1 : 0) + (block_removal_of_last_dq ? 1 : 0);
+    // TODO(yilyu): block_removal_of_first_dq is not functional, need to fix it
+    // TODO(yilyu): block_removal_of_last_dq is not functional, need to fix it
```
Comment (yufenglee) [inline onnxruntime/test/optimizer/qdq_transformer_test.cc:3679]: is it introduced by your PR? It works previously?

Comment (yihonglyu) [reply onnxruntime/test/optimizer/qdq_transformer_test.cc:3679]: After the application of `TransformerLevel::Level1`, the original model is as follows:
not_block_first_last
When `block_removal_of_first_dq` and `block_removal_of_last_dq` are applied, the test becomes:
block_first_last
With the introduction of this patch, both the first and last 'dq' are removed. I have yet to find a method that prevents their removal with this optimization. We might consider just removing the flag in the test case.

**Review by yufenglee — APPROVED** [review]
